code/go/pkg/validator/validator_test.go

```diff
@@ -1,13 +1,43 @@
 package validator
 
 import (
+	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/package-spec/code/go/internal/validator"
 )
 
 func TestValidate(t *testing.T) {
-	pkgRootPath := "../../internal/validator/test/packages/good"
-	errs := ValidateFromPath(pkgRootPath)
-	require.NoError(t, errs)
+	tests := map[string]struct {
+		expectedErrContains []string
+	}{
+		"good": {},
+		"bad_deploy_variants": {
+			expectedErrContains: []string{
```

Comment:
Such implementation verifies the expected error messages, but it doesn't relate them with particular files. It's vulnerable to pass the test if validation fails for any file.

Reply:
Added in 315cc073fd596651867f7c39ff0e4fa39d6c7262.